components/Navigation.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div class="flex ml-auto gap-10 m-2 mr-20 space-x-1 text-l">
+    <div
+      v-for="(page, i) in pages"
+      :key="i"
+      class="space-y-0.5"
+      @mouseover="page.hover = true"
+      @mouseleave="page.hover = false"
+    >
+      <!-- main button -->
+      <NuxtLink :to="page.parent.url">
+        <button
+          class="w-32 bg-blue-300 hover:bg-blue-500 text-white py-2 px-4 rounded"
+        >
+          {{ page.parent.title }}
+        </button>
+      </NuxtLink>
+      <!-- dropdown pages -->
+      <div
+        v-if="page.hover"
+        class="absolute flex flex-col space-y-0.5"
+      >
+        <div
+          v-for="subPage in page.children"
+          :key="subPage"
+        >
+          <NuxtLink :to="subPage.url">
+            <button class="w-32 bg-blue-300 hover:bg-blue-500 text-white py-2 px-4 rounded">
+              {{ subPage.title }}
+            </button>
+          </NuxtLink>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  data () {
+    return {
+      hover: false,
+      pages: [
+        {
+          parent: { title: 'CPM', url: '/cpm' },
+          children: [{ title: 'Explore', url: '/cpm/explore' },
```

Comment:
The few trees I've recently encountered seem to follow a structure along these lines:

```
pages: [
  { 
    title: 'CPM', 
    url: '/cpm', 
    children: [
      {
        title: 'Explore',
        url: 'cpm/explore',
        children: {}
      },
},
```

Note that having the `parent` attribute is not necessary. By leaving it out, the children become identical to the parent, which makes it possible to make endlessly nested trees that can be traversed tree recursively, e.g. like so:

```js
function traverse (tree) {
    if (tree.children != {}) {
      tree.forEach(node => traverse(node.children))
    } else {
      // This is a leaf node.
    }
}
```

Reply:
Looks really awesome! Just removed the redundant `parent` attribute.